sonar-plugin-api/src/main/java/org/sonar/api/ce/posttask/PostProjectAnalysisTaskTester.java

```diff
@@ -0,0 +1,530 @@
+    private static final String NAME_CAN_NOT_BE_NULL = "name can not be null";
+    private String uuid;
+    private String key;
+    private String name;
+
+    private ProjectBuilder() {
+      // prevents instantiation outside PostProjectAnalysisTaskTester
+    }
+
+    public ProjectBuilder setUuid(String uuid) {
+      this.uuid = requireNonNull(uuid, UUID_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ProjectBuilder setKey(String key) {
+      this.key = requireNonNull(key, KEY_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public ProjectBuilder setName(String name) {
+      this.name = requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public Project build() {
+      requireNonNull(uuid, UUID_CAN_NOT_BE_NULL);
+      requireNonNull(key, KEY_CAN_NOT_BE_NULL);
+      requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      return new Project() {
+        @Override
+        public String getUuid() {
+          return uuid;
+        }
+
+        @Override
+        public String getKey() {
+          return key;
+        }
+
+        @Override
+        public String getName() {
+          return name;
+        }
+
+        @Override
+        public String toString() {
+          return "Project{" +
+            "uuid='" + uuid + '\'' +
+            ", key='" + key + '\'' +
+            ", name='" + name + '\'' +
+            '}';
+        }
+
+      };
+    }
+  }
+
+  @Beta
+  public static final class QualityGateBuilder {
+    private static final String ID_CAN_NOT_BE_NULL = "id can not be null";
+    private static final String NAME_CAN_NOT_BE_NULL = "name can not be null";
+    private static final String STATUS_CAN_NOT_BE_NULL = "status can not be null";
+
+    private String id;
+    private String name;
+    private QualityGate.Status status;
+    private final List<QualityGate.Condition> conditions = new ArrayList<>();
+
+    private QualityGateBuilder() {
+      // prevents instantiation outside PostProjectAnalysisTaskTester
+    }
+
+    public QualityGateBuilder setId(String id) {
+      this.id = requireNonNull(id, ID_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public QualityGateBuilder setName(String name) {
+      this.name = requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public QualityGateBuilder setStatus(QualityGate.Status status) {
+      this.status = requireNonNull(status, STATUS_CAN_NOT_BE_NULL);
+      return this;
+    }
+
+    public QualityGateBuilder add(QualityGate.Condition condition) {
+      conditions.add(requireNonNull(condition, "condition can not be null"));
+      return this;
+    }
+
+    public QualityGateBuilder clearConditions() {
+      this.conditions.clear();
+      return this;
+    }
+
+    public QualityGate build() {
+      requireNonNull(id, ID_CAN_NOT_BE_NULL);
+      requireNonNull(name, NAME_CAN_NOT_BE_NULL);
+      requireNonNull(status, STATUS_CAN_NOT_BE_NULL);
+
+      return new QualityGate() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this anonymous class number of lines from 31 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1188)